### PR TITLE
Creating src dir for System.Diagnostics.StackTrace

### PR DESCRIPTION
**Pull request by dsgouda**
@weshaggard

### CODE REVIEW

**Comment by weshaggard** [comment]
@noahfalk please take a look.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.CoreCLR.csproj [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.CoreCLR.csproj]: You look like you accidentally committed the .CoreCLR and .Desktop project files.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.builds [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
+    <Project Include="System.Diagnostics.StackTrace.csproj">
+      <AdditionalProperties>TargetGroup=</AdditionalProperties>
+    </Project>
+    <Project Include="System.Diagnostics.StackTrace.csproj">
+      <AdditionalProperties>TargetGroup=netcore50aot</AdditionalProperties>
+    </Project>
+    <Project Include="System.Diagnostics.StackTrace.csproj">
+      <AdditionalProperties>TargetGroup=netcore50</AdditionalProperties>
+    </Project>
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.builds]: Looks like you do need at net46 version of this, which maps to the .Desktop project.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.builds [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.builds]: Please only have the TargetsWndows condition on all items but the TargetGroup=<empty> item.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <ExcludeDefaultReferences>true</ExcludeDefaultReferences>
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj]: You don't need ExcludeDefaultReferences.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    <TargetingPackReference Include="System.Private.Reflection" />
+    <ProjectReference Include="..\..\System.Runtime\src\System.Runtime.csproj">
+      <AdditionalProperties>TargetGroup=$(TargetGroup)</AdditionalProperties>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+     <Compile Include="System\Diagnostics\StackFrameExtensions.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)'!='netcore50aot'">
+    <Compile Include="System\Diagnostics\StackFrameExtensions.CoreCLR.cs" />
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj]: Update the extension of this file to match the new convention. For this one I think it should be no special extension.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)'=='netcore50aot'">
+    <Compile Include="System\Diagnostics\StackTrace.NetNative.cs" />
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj]: Update the extensions for these files to match the target group.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)'!='netcore50aot'">true</IsPartialFacadeAssembly>
+    <AssemblyVersion>4.0.1.0</AssemblyVersion>
+  </PropertyGroup>
+  <!-- Default configurations to help VS understand the options -->
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50_Release|AnyCPU'" />
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj]: I think you can drop netcore50 as it is redundant with the dnxcore50/default.

File: src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Release|AnyCPU'" />
+
+  <PropertyGroup Condition="'$(TargetGroup)'=='netcore50aot'">
+    <ProjectJson>netcore50aot\project.json</ProjectJson>
```
Comment (weshaggard) [inline src/System.Diagnostics.StackTrace/src/System.Diagnostics.StackTrace.csproj]: Given you are going to drop the netcore50 you should no longer need a separate project.json file.

**Comment by dsgouda** [comment]
@weshaggard could you have another look before I create a PR at dotnet master

**Comment by weshaggard** [comment]
Looks reasonable go ahead and submit PR to corefx master.

**Comment by dsgouda** [comment]
Review completed, opening PR on dotnet/master

**Pull request closed without merging**